app/Http/Controllers/Network/InternetController.php

```diff
@@ -4,250 +4,68 @@
-
-            foreach (User::admins() as $admin) {
-                Mail::to($admin)->send(new MacNeedsApproval($admin->name, user()->name));
-            }
-        }
-
-        MacAddress::create([
-            'user_id' => $target_id,
-            'state' => $state,
-            'comment' => $request->input('comment'),
-            'mac_address' => $request->input('mac_address'), //TODO use mutator
-        ]);
-
-        return redirect()->back()->with('message', __('general.successfully_added'));
-    }
-
-    public function getWifiConnectionsAdmin(Request $request): LengthAwarePaginator
-    {
-        $this->authorize('viewAny', WifiConnection::class);
-
-        $paginator = TabulatorPaginator::from(
-            WifiConnection::query()
-                ->groupBy(['wifi_username', 'mac_address', 'ip', 'lease_start', 'lease_end', 'note'])
-                ->select(['wifi_username', 'mac_address', 'ip', 'lease_start', 'lease_end', 'note', DB::raw('COUNT(*) as radius_connections')])
-        )->sortable(['wifi_username', 'mac_address', 'ip', 'lease_start'])
-            ->filterable(['wifi_username', 'mac_address', 'ip', 'lease_start'])
-            ->paginate();
-
-        return $paginator;
-    }
+        $this->authorize('handle', $internetAccess);
 
-    public function translateStates(): \Closure
-    {
-        return function ($data) {
-            $data->state = strtoupper($data->state);
-            $data->_state = $data->state;
-            switch ($data->state) {
-                case MacAddress::APPROVED:
-                    $data->state = __('internet.approved');
-                    break;
-                case MacAddress::REJECTED:
-                    $data->state = __('internet.rejected');
-                    break;
-                case MacAddress::REQUESTED:
-                    $data->state = __('internet.requested');
-                    break;
-            }
+        $internetAccess->resetPassword();
 
-            return $data;
-        };
+        return redirect()->back();
     }
 
     /**
      * Sends an email to all admins with the report of a fault.
      *
      * @param Request $request
-     * @return \Illuminate\Http\RedirectResponse
+     * @return RedirectResponse
+     * @throws AuthenticationException
      */
     public function reportFault(Request $request): RedirectResponse
     {
-        $validator = Validator::make($request->all(), [
+        $request->validate([
             'report' => 'required|string',
             'user_os' => 'required|string',
         ]);
-        $validator->validate();
 
-        foreach (User::withRole(Role::SYS_ADMIN)->get() as $admin) {
-            Mail::to($admin)->queue(new InternetFault($admin->name, user()->name, $request->report, $request->user_os));
+        foreach (User::admins() as $admin) {
+            Mail::to($admin)->queue(new InternetFault($admin->name, user()->name, $request->get('report'), $request->get('user_os')));
```

Comment:
Question for all of these kind of emails. Do we really need the emails stating the name of the admin? Couldn't it just say 'Dear Admin, ...'? If that was okay we could use `Mail::to(User::admins())->...` instead.